controllers/work_controller.go

```diff
@@ -124,5 +127,30 @@ func (r *WorkReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&v1alpha1.Work{}).
 		Owns(&v1alpha1.WorkPlacement{}).
+		Watches(
+			&v1alpha1.Destination{},
+			handler.EnqueueRequestsFromMapFunc(r.requestReconilationOfAllWorksOnDestinationCreateOrUpdate),
+		).
 		Complete(r)
 }
+
+func (r *WorkReconciler) requestReconilationOfAllWorksOnDestinationCreateOrUpdate(ctx context.Context, obj client.Object) []reconcile.Request {
+	dest := obj.(*v1alpha1.Destination)
+
+	if dest.GetDeletionTimestamp() != nil {
+		return nil
+	}
+
+	allWorks := &v1alpha1.WorkList{}
+	err := r.Client.List(ctx, allWorks)
```

Comment:
`scheduler.ReconcileAllDependencyWorks()` only reconciles if `work.IsDependency()` is true. Should we have the same check in here?

Reply:
I don't think so. If a work is for a resource request and is not scheduled (no matching destination) we do want to reconcile the work. Also if a Work for a resource request is scheduled to a destination, and then that destination is updated to no longer match the labels that made it get scheduled to by the work to begin with we update the work status.